scheduler/scheduler-node/src/main/java/org/ow2/proactive/resourcemanager/task/client/RMNodeClient.java

```diff
@@ -0,0 +1,651 @@
+import org.ow2.proactive.resourcemanager.frontend.topology.TopologyInfo;
+import org.ow2.proactive.resourcemanager.nodesource.common.NodeSourceConfiguration;
+import org.ow2.proactive.resourcemanager.nodesource.common.PluginDescriptor;
+import org.ow2.proactive.scheduler.common.exception.NotConnectedException;
+import org.ow2.proactive.scheduler.rest.IRMClient;
+import org.ow2.proactive.scripting.ScriptResult;
+import org.ow2.proactive_grid_cloud_portal.common.RMRestInterface;
+import org.ow2.proactive_grid_cloud_portal.common.dto.LoginForm;
+import org.ow2.proactive_grid_cloud_portal.rm.client.RMRestClient;
+import org.ow2.proactive_grid_cloud_portal.scheduler.exception.PermissionRestException;
+import org.ow2.proactive_grid_cloud_portal.scheduler.exception.RestException;
+
+
+@PublicAPI
+public class RMNodeClient implements IRMClient, Serializable {
+
+    private String schedulerRestUrl;
+
+    private CredData userCreds;
+
+    private RMRestInterface rm;
+
+    private ConnectionInfo connectionInfo;
+
+    private String sessionId;
+
+    public RMNodeClient(CredData userCreds, String schedulerRestUrl) {
+        this.schedulerRestUrl = schedulerRestUrl;
+        this.userCreds = userCreds;
+    }
+
+    public void connect() throws Exception {
+        init(new ConnectionInfo(schedulerRestUrl, userCreds.getLogin(), userCreds.getPassword(), null, true));
+    }
+
+    public void connect(String url) throws Exception {
+        schedulerRestUrl = url;
+        connect();
+    }
+
+    @Override
+    public void init(ConnectionInfo connectionInfo) throws Exception {
+        HttpClient client = new HttpClientBuilder().insecure(connectionInfo.isInsecure()).useSystemProperties().build();
+
+        RMRestClient restApiClient = new RMRestClient(connectionInfo.getUrl(), new ApacheHttpClient4Engine(client));
+
+        this.rm = restApiClient.getRm();
+
+        this.connectionInfo = connectionInfo;
+
+        renewSession();
+    }
+
+    public void renewSession() {
+        try {
+            String sid = rm.rmConnect(connectionInfo.getLogin(), connectionInfo.getPassword());
+            setSession(sid);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public ConnectionInfo getConnectionInfo() {
+        return connectionInfo;
+    }
+
+    @Override
+    public void setSession(String sid) {
+        this.sessionId = sid;
+    }
+
+    @Override
+    public String getSession() {
+        return sessionId;
+    }
+
+    /**
+     * Disconnects from resource manager and releases all the nodes taken by
+     * user for computations.
+     */
+    public void disconnect() throws NotConnectedException {
+        rm.rmDisconnect(sessionId);
+    }
+
+    /**
+     * @return Returns the state of the Resource Manager
+     */
+    public RMState getState() throws NotConnectedException {
+        return rm.getState(sessionId);
```

Comment:
what happens when the connect() did not succeed or was not called, what kind of exception is thrown?

Reply:
no exception. Methods return nulls.

Reply:
Then we need to find a way to handle when the user is not connected to the rm and throw exceptions if not

Reply:
ok. i will call checkSession() method in the beginning of each method